File: C/C++/112.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main(){
    int N;
    cin >> N;

    int minTime = 1e9;
    int maxTime = -1e9;

    for (int i = 0; i < N; i++){
        int s, t, f;
        cin >> s >> f >> t;
        int dayLength = s + f + (24 - t);

        minTime = min(minTime, dayLength);
        maxTime = max(maxTime, dayLength);
    }

    cout << minTime << endl;
    cout << maxTime << endl;
    return 0;
}

File: C/Java/112.java
import java.util.*;

public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        
        int minTime = Integer.MAX_VALUE;
        int maxTime = Integer.MIN_VALUE;

        for (int i = 0; i < N; i++){
            int s = sc.nextInt();
            int f = sc.nextInt();
            int t = sc.nextInt();
            int dayLength = s + f + (24 - t);
        

            minTime = Math.min(minTime, dayLength);
            maxTime = Math.max(maxTime, dayLength);
        }
    }

    System.out.println(minTime);
    System.out.println(maxTime);
}